src/manager/TickManager.ts

```diff
@@ -1,46 +1,190 @@
+    const sortedTicks = this.getSortedTicks();
+    const compressed = Math.floor(tick / tickSpacing); // matches rounding in the code
+
+    if (leftToRight) {
+      const wordPos = compressed >> 8;
+      const minimum = (wordPos << 8) * tickSpacing;
+
+      if (this.isBelowSmallest(sortedTicks, tick)) {
+        return { nextTick: minimum, initialized: false };
+      }
+
+      const index = this.nextInitializedTick(sortedTicks, tick, leftToRight)
+        .tickIndex;
+      const nextInitializedTick = Math.max(minimum, index);
+      return {
+        nextTick: nextInitializedTick,
+        initialized: nextInitializedTick === index,
+      };
+    } else {
+      const wordPos = (compressed + 1) >> 8;
+      const maximum = ((wordPos + 1) << 8) * tickSpacing - 1;
+
+      if (this.isAtOrAboveLargest(sortedTicks, tick)) {
+        return { nextTick: maximum, initialized: false };
+      }
+
+      const index = this.nextInitializedTick(sortedTicks, tick, leftToRight)
+        .tickIndex;
+      const nextInitializedTick = Math.min(maximum, index);
+      return {
+        nextTick: nextInitializedTick,
+        initialized: nextInitializedTick === index,
+      };
+    }
   }
 
   getFeeGrowthInside(
     tickLower: number,
     tickUpper: number,
     tickCurrent: number,
     feeGrowthGlobal0X128: JSBI,
-    feeGrowthGlobal1X128: JSBI,
-    leftToRight: boolean
-  ): JSBI {
-    // TODO
-    return JSBI.BigInt(0);
+    feeGrowthGlobal1X128: JSBI
+  ): { feeGrowthInside0X128: JSBI; feeGrowthInside1X128: JSBI } {
+    const lower = this.get(tickLower);
+    const upper = this.get(tickUpper);
+    let feeGrowthBelow0X128: JSBI;
+    let feeGrowthBelow1X128: JSBI;
+    if (tickCurrent >= tickLower) {
+      feeGrowthBelow0X128 = lower.feeGrowthOutside0X128;
+      feeGrowthBelow1X128 = lower.feeGrowthOutside1X128;
+    } else {
+      feeGrowthBelow0X128 = JSBI.subtract(
+        feeGrowthGlobal0X128,
+        lower.feeGrowthOutside0X128
+      );
+      feeGrowthBelow1X128 = JSBI.subtract(
+        feeGrowthGlobal1X128,
+        lower.feeGrowthOutside1X128
+      );
+    }
+
+    let feeGrowthAbove0X128: JSBI;
+    let feeGrowthAbove1X128: JSBI;
+    if (tickCurrent < tickUpper) {
+      feeGrowthAbove0X128 = upper.feeGrowthOutside0X128;
+      feeGrowthAbove1X128 = upper.feeGrowthOutside1X128;
+    } else {
+      feeGrowthAbove0X128 = JSBI.subtract(
+        feeGrowthGlobal0X128,
+        upper.feeGrowthOutside0X128
+      );
+      feeGrowthAbove1X128 = JSBI.subtract(
+        feeGrowthGlobal1X128,
+        upper.feeGrowthOutside1X128
+      );
+    }
+    return {
+      feeGrowthInside0X128: JSBI.subtract(
+        JSBI.subtract(feeGrowthGlobal0X128, feeGrowthBelow0X128),
+        feeGrowthAbove0X128
+      ),
+      feeGrowthInside1X128: JSBI.subtract(
+        JSBI.subtract(feeGrowthGlobal1X128, feeGrowthBelow1X128),
+        feeGrowthAbove1X128
+      ),
+    };
+  }
+
+  clear(tick: number) {
```

Comment:
Thank you and this method is necessary indeed for liquidity management. Let's label this pr with `breaking-change`.